Sources/Core/Other/WindowMonitor.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  WindowMonitor.swift
+//  XestiMonitors-iOS
+//
+//  Created by Martin Mungai on 20/03/2018.
+//
+//  © 2016 J. G. Pusey (see LICENSE.md)
+
+#if os(iOS)
+    import Foundation
+    import UIKit
+
+    ///
+    /// A `WindowMonitor` instance monitors the window for changes to its
+    /// visibility or to its frame.
+    ///
+    
+    public class WindowMonitor: BaseNotificationMonitor {
+        ///
+        /// Encapsulates changes to the visibility of the window and to the frame
+        /// of the window.
+        ///
+        public enum Event {
+            
+            case didBecomeVisible(UIWindow)
+            case didBecomeHidden(UIWindow)
+            case didBecomeKey(UIWindow)
+            case didResignKey(UIWindow)
+        }
+        
+        public struct Options: OptionSet {
+            
+            ///
+            /// Monitor `didBecomeVisible` events
+            ///
+            public static let didBecomeVisible = Options(rawValue: 1 << 0)
+            
+            
+            ///
+            /// Monitor `didBecomeHidden` events
+            ///
+            public static let didBecomeHidden = Options(rawValue: 1 << 1)
+            
+            ///
+            /// Monitor `didBecomeKey` events
+            ///
+            public static let didBecomeKey = Options(rawValue: 1 << 2)
+            
+            ///
+            /// Monitor `didResignKey` events
+            ///
+            public static let didResignKey = Options(rawValue: 1 << 3)
+            
+            ///
+            /// Monitor all events
+            ///
+            public static let all: Options = [.didBecomeVisible,
+                                              .didBecomeHidden,
+                                              .didBecomeKey,
+                                              .didResignKey]
+            
+            public init(rawValue: UInt) {
+                self.rawValue = rawValue
+            }
+            
+            public let rawValue: UInt
+        }
+        
+        ///
+        /// Initializes a new `WindowMonitor`.
+        ///
+        /// - Parameters:
+        ///   - queue:      The operation queue on which the handler executes. By
+        ///                 default, the main operation queue is used.
+        ///   - options:    The options that specify which events to monitor. By
+        ///                 default, all events are monitored.
+        ///   - handler:    The handler to call when the visibility of the keyboard
+        ///                 or the frame of the keyboard changes or is about to
+        ///                 change.
+        ///
+        
+        public init(queue: OperationQueue = .main,
```

Comment:
Parameter order should be: `window`, `options`, `queue`, `handler`.

Please update doc comment for parameters accordingly.